cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupPage/GroupPanel/WidgetCollection/WidgetAttributesPanel/index.tsx

```diff
@@ -71,14 +73,34 @@ const WidgetAttributesPanelView: React.FC<IWidgetAttributesPanelProps> = ({
   // 'widgetToAttributes' will only be changed when user clicks on 'save' button or closes the dialog.
   const [localWidgetToAttributes, setLocalWidgetToAttributes] = React.useState(widgetToAttributes);
 
+  const [loading, setLoading] = React.useState(false);
+
+  // When widget attributes changes, show loading view for 500ms
+  // This is in order to force rerender WidgetAttributeInput component
+  const debouncedUpdate = debounce(() => {
+    setLoading(false);
+  }, 500);
+
+  React.useEffect(() => {
+    // after a setTimeout for 500ms, set the loading state back to false
+    setLoading(true);
+    setLocalWidgetToAttributes(widgetToAttributes);
+    debouncedUpdate();
```

Comment:
this is a wrong usage for debounce, this should be a timeout.

You should debounce the entire callback of useEffect, and inside it have a timeout to set the loading to false